ledger/participant-integration-api/src/main/scala/platform/store/serialization/Compression.scala

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.serialization
+
+import java.io.{ByteArrayOutputStream, InputStream, OutputStream}
+import java.util.zip.{GZIPInputStream, GZIPOutputStream}
+
+import com.daml.platform.store.dao.events.CompressionMetrics
+
+private[platform] object Compression {
+
+  sealed abstract class Algorithm(val id: Option[Int]) {
+    final def compress(
+        uncompressed: Array[Byte],
+        metrics: CompressionMetrics.Field,
+    ): Array[Byte] = {
+      val output = new ByteArrayOutputStream(uncompressed.length)
+      val gzip = compress(output)
+      try {
+        gzip.write(uncompressed)
+      } finally {
```

Comment:
@mzagorski-da : this swallows an exception and then just pretends everything was allright doesn't it?

That sounds like a bad problem in waiting. Can you clarify how this code is supposed to work in that case?

Reply:
@meiersi-da I'm confused, where is the exception swallowed?

Reply:
o.O -- I'm outing myself as a Java noob here :/ -- please ignore my comment. `finally` obviously lets the exception bubble up.

:see_no_evil: